### PR TITLE
Feature/month paginate

### PR DESCRIPTION
**Pull request by dwoznicka**

### CODE REVIEW

File: client/js/components/pages/month-paginator.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    e.preventDefault();
+    this.props.onClickYear(year);
+  }
+
+  render() {
+    let p = this.props;
+    let pagesArray = Array.apply(null, Array(12)).map(function (x, i) { return i; });
+    let currentYear = p.currentYear;
+    let monthNames = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
+    let realDate = new Date(Date.now());
```
Comment (vrael560) [inline client/js/components/pages/month-paginator.jsx]: `new Date()` return current date without arguments.

File: server/routes/snippets.js [hunk 2]
```diff
@@ -11,21 +11,26 @@ var authChecker = function(req, res, next) {
...
+  /* January is 0, December is 11 */
+  var options = { where: {}, order: [] };
+  var now = new Date(Date.now());
+  var month = parseInt(req.query.month);
+  var year = parseInt(req.query.year);
+  var first = new Date(year, month, 1);
+  var last = new Date(year, month+1, 0);
+  options.where = { createdAt: { $gte: first, $lte: last } };
+  
+  options.order = (now.getFullYear() == year && now.getMonth() == month) ? [['createdAt', 'DESC']] : [ ['avg', 'DESC'], ['createdAt', 'DESC'] ];
```
Comment (vrael560) [inline server/routes/snippets.js]: I think we've agreed that snippets index list will be simple ordered by creation date and there will be other button/select box to get snippets by `avg`. So index action should take two sets of parameters one for pagination and second for sort order. Am I right @blimusiek?

File: server/test/snippet.test.js [hunk 3]
```diff
@@ -71,7 +75,7 @@ describe('Snippets routes', function() {
 
     it('should return array of snippets', function (done) {
       request(app)
-        .get('/snippets')
+        .get('/snippets?year='+year+'&month='+month)
```
Comment (vrael560) [inline server/test/snippet.test.js]: I won't insist but we have this feature available in node 4 :+1: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

**Comment by vrael560** [comment]
@blimusiek fine for me :+1:

File: client/js/components/pages/month-paginator.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  _handleClickYear(year, e){
+    e.preventDefault();
+    this.props.onClickYear(year);
+  }
+
+  render() {
+    let p = this.props;
+    let pagesArray = Array.apply(null, Array(12)).map(function (x, i) { return i; });
```
Comment (blimusiek) [inline client/js/components/pages/month-paginator.jsx]: why not `[0...12]`

File: server/routes/snippets.js [hunk 5]
```diff
@@ -11,21 +11,24 @@ var authChecker = function(req, res, next) {
...
+/* GET snippets from specified month */
 router.get('/', function (req, res) {
-  var perPage = req.query.results;
-  var page = req.query.offset;
+  /* January is 0, December is 11 */
+  var options = { where: {}, order: [] };
+  var month = parseInt(req.query.month);
+  var year = parseInt(req.query.year);
+  var first = new Date(year, month, 1);
+  var last = new Date(year, month+1, 0);
```
Comment (blimusiek) [inline server/routes/snippets.js]: cool

**Pull request merged**